keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -772,6 +776,13 @@ public double determinant() {
         return new LUDecomposition(matrix).getDeterminant();
     }
 
+    @Override
+    public DoubleTensor tad(int dimension, int index) {
+        INDArray dup = tensor.dup();
+        INDArray tad = dup.slice(index, dimension);
```

Comment:
Is there any reason why our API puts the arguments the other way round? It looks dangerous since they're both of the same type so they could get mixed up.